### PR TITLE
Mailer setup

### PR DESCRIPTION
**Pull request by bazarnyi**
- SMTP set up
- User email confirmation upon registration added

[Trello link #1](https://trello.com/c/Gws7ZTio/13-module-2-%D0%BD%D0%B0%D1%81%D1%82%D1%80%D0%BE%D0%B8%D1%82%D1%8C-smtp-%D1%81%D0%B5%D1%80%D0%B2%D0%B5%D1%80-%D0%B8-%D1%80%D0%B0%D1%81%D1%81%D1%8B%D0%BB%D0%BA%D1%83-%D0%BF%D0%B8%D1%81%D0%B5%D0%BC-%D1%80%D0%B5%D0%B3%D0%B8%D1%81%D1%82%D1%80%D0%B0%D1%86%D0%B8%D0%B8)
[Trello link #2](https://trello.com/c/UAdbc8hr/14-module-2-%D0%BD%D0%B0%D1%81%D1%82%D1%80%D0%BE%D0%B8%D1%82%D1%8C-%D1%81%D0%B1%D1%80%D0%BE%D1%81-%D0%BF%D0%B0%D1%80%D0%BE%D0%BB%D1%8F-%D0%B4%D0%BB%D1%8F-devise-user)

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 class User < ApplicationRecord
   # Include default devise modules. Others available are:
-  # :confirmable, :lockable, :timeoutable and :omniauthable
+  # :lockable, :timeoutable and :omniauthable
```
Comment (atatarenko) [inline app/models/user.rb:3]: What do you think about getting rid of unused commented code?

File: config/environments/production.rb [hunk 2]
```diff
@@ -91,4 +91,22 @@
 
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
+
+  config.action_mailer.default_url_options = { :host => 'https://den-budget-planner.herokuapp.com/' }
```
Comment (atatarenko) [inline config/environments/production.rb]: I suggest new hash syntax here
```ruby
config.action_mailer.default_url_options = { host: 'https://den-budget-planner.herokuapp.com/' }
```

**Review by atatarenko — APPROVED** [review]
LGTM with suggestions

Could you deploy the app on the Heroku so I am able to make a smoke tests?

File: app/models/user.rb (hunk 1 above)
Comment (bazarnyi) [reply app/models/user.rb:3]: I would leave in for now. Will see how I would implement omniauth,Then - yes, I will get rid of it

File: config/environments/production.rb (hunk 2 above)
Comment (bazarnyi) [reply config/environments/production.rb]: Sure will do

**Comment by bazarnyi** [comment]
@atatarenko The app from the current branch was on Heroku (and still there) at the moment of creating PR. 
It still doesn't have any UI navigation elements (will be done in next PR). So the only way of testing is directly navigating through URLs. Not very user-friendly but working for now. 

After adding more UI I would definitely add Capybara Features to test it

**Pull request merged**